yt_dlp/extractor/lsm.py

```diff
@@ -0,0 +1,305 @@
+                'duration': 3300,
+            },
+        }],
+    }, {
+        'url': 'https://radioteatris.lsm.lv/lv/embed/?id=&show=1269&theme=white&size=16x9',
+        'md5': '24810d4a961da2295d9860afdcaf4f5a',
+        'info_dict': {
+            'id': 'a230690',
+            'ext': 'mp3',
+            'title': 'Jens Ahlboms "Spārni". Radioizrāde ar Mārtiņa Freimaņa mūziku',
+            'thumbnail': 'https://radioteatris.lsm.lv/public/assets/shows/62f13023a457c.jpg',
+            'duration': 1788,
+        }
+    }, {
+        'url': 'https://lr1.lsm.lv/lv/embed/?id=166557&show=0&theme=white&size=16x9',
+        'md5': '5d5e191e718b7644e5118b7b4e093a6d',
+        'info_dict': {
+            'id': 'a303104',
+            'ext': 'mp4',
+            'thumbnail': 'https://pic.latvijasradio.lv/public/assets/media/c/5/gallery_a83ad2c2.jpg',
+            'title': 'Krustpunktā Lielā intervija: Valsts prezidents Egils Levits',
+            'duration': 3222,
+        }
+    }, {
+        'url': 'https://lr1.lsm.lv/lv/embed/?id=183522&show=0&theme=white&size=16x9',
+        'only_matching': True,
+    }, {
+        'url': 'https://lr2.lsm.lv/lv/embed/?id=182126&show=0&theme=white&size=16x9',
+        'only_matching': True,
+    }, {
+        'url': 'https://klasika.lsm.lv/lv/embed/?id=110806&show=0&theme=white&size=16x9',
+        'only_matching': True,
+    }, {
+        'url': 'https://lr4.lsm.lv/lv/embed/?id=184282&show=0&theme=white&size=16x9',
+        'only_matching': True,
+    }, {
+        'url': 'https://pieci.lv/lv/embed/?id=168896&show=0&theme=white&size=16x9',
+        'only_matching': True,
+    }, {
+        'url': 'https://naba.lsm.lv/lv/embed/?id=182901&show=0&theme=white&size=16x9',
+        'only_matching': True,
+    }, {
+        'url': 'https://radioteatris.lsm.lv/lv/embed/?id=176439&show=0&theme=white&size=16x9',
+        'only_matching': True,
+    }, {
+        'url': 'https://lr1.lsm.lv/lv/pleijeris/?embed=0&id=48205&time=00%3A00&idx=0',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        query = parse_qs(url)
+        video_id = traverse_obj(query, ('show', 0, {int_or_none}))
+
+        if video_id is None or video_id == 0:
+            video_id = traverse_obj(query, ('id', 0, {int_or_none}))
+
+        webpage = self._download_webpage(url, video_id)
+
+        player_data, media_data = self._html_search_regex(
+            r'LR\.audio\.Player\s*\([^{]*(?P<player>\{.*?\}),(?P<media>\{.*\})\);',
+            webpage, 'player json', group=('player', 'media'))
+
+        player_json = self._parse_json(player_data, video_id, js_to_json)
+        media_json = self._parse_json(media_data, video_id, js_to_json)
+
+        entries = []
+        for i, audio_json in enumerate(traverse_obj(media_json, ('audio', ...))):
+            formats = []
+            for source_url in traverse_obj(media_json, (('audio', 'video'), i, 'sources', ..., 'file', {url_or_none})):
```

Comment:
In all examples I was able to find, `video` contained standalone videos (i.e., with both audio and video track), while the `audio` contained audio only MP3 versions.
In the test case provided here, the audio only MP3 version is even a different length than the audio track of the m3u8, providing another reason for why these should be split into different entries.

Reply:
My biggest problem with treating the audio and video as separate is that in the provided example, the video dict has an empty title, whereas you can get the title from the audio version.
Also while they may have different actual lengths, the audio and video versions are effectively the same thing, only difference is that the video has the intro repeated at the end.

What would you say to something like:
```python
def get_entry_ids(json):
    result = []

    for item_id in traverse_obj(json, (('audio', 'video'), ..., 'id')):
        id = item_id[1:]
        if id not in result:
            result.append(id)

    return result

def find_entry_by_id(json, id):
    result = []

    for item in traverse_obj(json, (('audio', 'video'), ...)):
        if item['id'][1:] == id:
            result.append(item)

    return result

entries = []
for id in get_entry_ids(media_json):
    formats = []
    entry = find_entry_by_id(media_json, id)

    for source_url in traverse_obj(entry, (..., 'sources', ..., 'file', {url_or_none})):
        if source_url.endswith('.m3u8'):
            formats.extend(self._extract_m3u8_formats(source_url, video_id))
        else:
            formats.append({'url': source_url})

    entries.append({
        'id': id,
        'formats': formats,
        'thumbnail': urljoin(url, player_json.get('poster')),
        **traverse_obj(entry, (..., {
            'title': 'title',
            'duration': ('duration', {int_or_none})
        }), get_all=False)
    })
```
This would still work even if a video exists without an audio version and if the order of them is not consistent.

Reply:
Simpler way to search:

```py
data = {}
for entry in traverse_obj(media_json, ('video', lambda x: x['id'])):
    data[entry['id'][1:]] = [entry, None]
for entry in traverse_obj(media_json, ('audio', lambda x: x['id'])):
    data.setdefault(entry['id'][1:], [None, None])[1] = entry

for id_, video, audio in data:
    ...
```

Audio-only and video-only formats must have `vcodec`/`acodec`=`'none'` respectively. I've kept audio and video separate in the final structure so we can set these fields. If there's another way to detect that, it may be even simpler.

Reply:
Looking around, I found some entries with vastly different runtimes (and different content?) between the audio and video version. https://lr4.lsm.lv/lv/embed/?id=166309&show=0&theme=white&size=16x9 15 vs. 55 minutes in this case.
This one specifically could also be a technical issue on their end, not quite sure, but there are also a fair number of other entries with 5-10 minute differences in runtime.

I feel like it doesn't make sense to have content this different be listed as different formats.
How about having it be two videos, but if there is no title set on the video track, the title get's pulled from the audio track with the same ID and ` - Video Version` appended?